crates/consensus/src/signed.rs

```diff
@@ -57,7 +75,16 @@ impl<T: SignableTransaction<Signature>> Signed<T, Signature> {
     pub fn recover_signer(
         &self,
     ) -> Result<alloy_primitives::Address, alloy_primitives::SignatureError> {
+        #[cfg(not(feature = "no_std"))]
+        if let Some(signer) = self.signer() {
+            return Ok(*signer);
+        }
```

Comment:
proposed api

- delete `recover_signer` as standard API
- add `new_with_unchecked_signer` for instantiation
- modify `signer` as follows:

```

/// get the signer if known or recoverable
fn signer(&self) -> Option<Address> {
    #[cfg(not(feature = "k256"))]
    self.signer.get().copied()

    #[cfg(feature = "k256")]
    self.signer.get_or_init(|| { /* recovery logic here */ });
}

```

Reply:
also ok if implemented as 

```
#[cfg(not(feature = "k256"))]
fn signer(&self) -> Option<Address> { ... }

#[cfg(feature = "k256")]
fn signer(&self) -> Option<Address> { ... }
```

whatever feels cleaner